File: StoreSearchString.py
database = {}
key = input().strip()

while key != "*":
    database[key] = True
    key = input().strip()

while True:
    cmd, key = input().strip().split()
    if cmd == "find":
        if key in database:
            print("1")
        else:
            print("0")
    elif cmd == "insert":
        if key in database:
            print("0")
        else:
            database[key] = True
            print("1")
    if key == "***":
        break

File: StoreSearchString.cpp
#include <iostream>
#include <unordered_map>
#include <string>

using namespace std;

int main()
{
    unordered_map<string, bool> database;
    string key;

    while (cin >> key)
    {
        if (key == "*")
            break;
        database[key] = true;
    }

    string cmd;
    while (cin >> cmd >> key)
    {
        if (cmd == "find")
        {
            if (database.find(key) != database.end())
            {
                cout << "1\n"; // check xem giá trị có trong database không
            }
            else
            {
                cout << "0\n";
            }
        }
        else if (cmd == "insert")
        {
            if (database.find(key) != database.end())
            {
                cout << "0\n";
            }
            else
            {
                database[key] = true; // thêm phần tử vào database
                cout << "1\n";
            }
        }
        if (key == "***")
            break;
    }

    return 0;
}
